tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -403,6 +404,158 @@ def _query_rules(self, resource):
+     1000 rules (maximum tested rule count).
+
+     .. code-block:: yaml
+
+         - type: set-firewall-rules
+               ip-rules:
+                   - 11.12.13.0/16
+                   - ServiceTags.AppService.CentralUS
+
+
+     :example:
+
+     Find CosmosDB accounts without any firewall rules.
+
+     Enable the firewall and allow:
+     - All Azure Cloud IP space
+     - All Portal UI IP space
+     - Two additional external IP ranges
+
+     Mark ``append: True`` to ensure we only add to the existing configuration
+     which in this case means we don't remove any previously configured
+     vnet firewall rules.
+
+     .. code-block:: yaml
+
+        policies:
+          - name: cosmos-firewall
+            resource: azure.cosmosdb
+            filters:
+              # The firewall is disabled
+              - type: value
+                key: properties.ipRangeFilter
+                value: empty
+            actions:
+              - type: set-firewall-rules
+                append: True
+                bypass-rules:
+                  - AzureCloud
+                  - Portal
+                ip-rules:
+                  - 19.0.0.0/16
+                  - 20.0.1.2
+
+
+     Cosmos firewalls are disabled by simply configuring them with empty values.
+     We can do this with an empty action, which defaults to ``append: False``.
+
+     .. code-block:: yaml
+
+        policies:
+          - name: cosmos-firewall-clear
+            resource: azure.cosmosdb
+            filters:
+              # The firewall is enabled
+              - not:
+                - type: value
+                  key: properties.ipRangeFilter
+                  value: empty
+            actions:
+              - type: set-firewall-rules
+
+
+     """
+    schema = type_schema(
+        'set-firewall-rules',
+        required=[],
+        **{
+            'append': {'type': 'boolean', "default": False},
+            'bypass-rules': {'type': 'array', 'items': {
+                'enum': ['Portal', 'AzureCloud']}},
+            'ip-rules': {'type': 'array', 'items': {'type': 'string'}},
+            'virtual-network-rules': {'type': 'array', 'items': {'type': 'string'}}
+        }
+    )
+
+    def __init__(self, data, manager=None):
+        super(CosmosSetFirewallAction, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.cosmosdb')
+        self.rule_limit = 1000
+
+    def _process_resource(self, resource):
+        existing_ip = filter(None, resource['properties'].get('ipRangeFilter', '').split(','))
+        rules = self._build_ip_rules(existing_ip, self.data.get('ip-rules', []))
+
+        # Cosmos DB does not have real bypass
+        # instead the portal UI adds these values to your
+        # rules filter when you check the box.
+        bypass = self.data.get('bypass-rules', [])
+        if 'Portal' in bypass:
+            rules.extend(['104.42.195.92',
```

Comment:
I assume there is no docs\published list of these IPs?

Reply:
Not only that, but there is no reasonable way for them to even handle them changing - what will they do, put the checkbox in the portal into a tri-state mode if they detect an old version of the IP list and not a new one? :: sigh :: 

Not really our problem, as long as we do what the portal does I guess.

Reply:
How did you get this list of IP addresses if not documented? Maybe your repro steps would help here so if this list ever needs editing the next dev knows how to get them.

Reply:
Or is that the comment above that it's in the portal?

Reply:
Is the list always the same for all tenants/subscriptions/etc?

Reply:
Yeah they just hard code the portal webserver edges I believe.  Per comment its just from what the portal added when I checked the box there.